File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/TraceLevels.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

final class TraceLevels {
    TraceLevels(Properties properties) {
        dispatchCat = "Dispatch";
        networkCat = "Network";
        protocolCat = "Protocol";
        retryCat = "Retry";
        locationCat = "Locator";
        slicingCat = "Slicing";
        threadPoolCat = "ThreadPool";

        final String keyBase = "Ice.Trace.";

        dispatch = properties.getPropertyAsInt(keyBase + dispatchCat);
        network = properties.getPropertyAsInt(keyBase + networkCat);
        protocol = properties.getPropertyAsInt(keyBase + protocolCat);
        retry = properties.getPropertyAsInt(keyBase + retryCat);
        location = properties.getPropertyAsInt(keyBase + locationCat);
        slicing = properties.getPropertyAsInt(keyBase + slicingCat);
        threadPool = properties.getPropertyAsInt(keyBase + threadPoolCat);
    }

    public final int dispatch;
    public final String dispatchCat;
    public final int network;
    public final String networkCat;
    public final int protocol;
    public final String protocolCat;
    public final int retry;
    public final String retryCat;
    public final int location;
    public final String locationCat;
    public final int slicing;
    public final String threadPoolCat;
    public final int threadPool;
    public final String slicingCat;
}

File: csharp/src/Ice/Internal/TraceLevels.cs
// Copyright (c) ZeroC, Inc.

namespace Ice.Internal;

public sealed class TraceLevels
{
    internal TraceLevels(Ice.Properties properties)
    {
        dispatchCat = "Dispatch";
        networkCat = "Network";
        protocolCat = "Protocol";
        retryCat = "Retry";
        locationCat = "Locator";
        slicingCat = "Slicing";
        threadPoolCat = "ThreadPool";

        string keyBase = "Ice.Trace.";

        dispatch = properties.getIcePropertyAsInt(keyBase + dispatchCat);
        network = properties.getIcePropertyAsInt(keyBase + networkCat);
        protocol = properties.getIcePropertyAsInt(keyBase + protocolCat);
        retry = properties.getIcePropertyAsInt(keyBase + retryCat);
        location = properties.getIcePropertyAsInt(keyBase + locationCat);
        slicing = properties.getIcePropertyAsInt(keyBase + slicingCat);
        threadPool = properties.getIcePropertyAsInt(keyBase + threadPoolCat);
    }

    public readonly int dispatch;
    public readonly string dispatchCat;
    public readonly int network;
    public readonly string networkCat;
    public readonly int protocol;
    public readonly string protocolCat;
    public readonly int retry;
    public readonly string retryCat;
    public readonly int location;
    public readonly string locationCat;
    public readonly int slicing;
    public readonly string slicingCat;
    public readonly int threadPool;
    public readonly string threadPoolCat;
}
